packages/lodestar/src/db/repositories/blockArchive.ts

```diff
@@ -101,14 +101,21 @@ export class BlockArchiveRepository extends Repository<Slot, allForks.SignedBeac
     ]);
   }
 
+  async *p2pBlockStream(opts?: IBlockFilterOptions): AsyncIterable<P2pBlockResponse> {
+    const firstSlot = this.getFirstSlot(opts);
+    const binaryValuesStream = super.binaryValuesStream(opts);
+    const step = (opts && opts.step) || 1;
+
+    for await (const bytes of binaryValuesStream) {
+      const slot = getSlotFromBytes(bytes);
+      if ((slot - firstSlot) % step === 0) {
+        yield {bytes, slot};
```

Comment:
The reasoning for attaching the slot here is to get the fork latter when constructing the response? Or is there other usages

Reply:
@dapplion  it's just to get the fork later